app/src/rooms/views.py

```diff
@@ -57,66 +58,31 @@ def index(request):
     return render(request, 'rooms/index.html', context)
 
 
-# GET
-@vary_on_cookie
-@login_required
+@staff_member_required
 @require_http_methods(['GET'])
-def status(request):
-    # Ajax
-    # Return JSON view of rooms
-    zosia = get_object_or_404(Zosia, active=True)
-    user_prefs = get_object_or_404(UserPreferences, zosia=zosia, user=request.user)
-    can_start_rooming = zosia.can_user_choose_room(user_prefs)
-    rooms = Room.objects.all_visible().select_related('lock').prefetch_related('members').all()
-    rooms_view = []
-
-    for room in rooms:
-        dic = room_to_dict(room)
-        dic['is_owned_by'] = room.is_locked and room.lock.is_owned_by(
-            request.user) and room.lock.password
-        dic['people'] = list(map(user_to_dict, room.members.all()))
-        dic['inside'] = request.user.pk in map(lambda x: x.pk, room.members.all())
-        rooms_view.append(dic)
-
-    view = {
-        'can_start_rooming': can_start_rooming,
-        'rooms': rooms_view,
-    }
-    return JsonResponse(view)
-
+def list_by_user(request):
+    prefs = UserPreferences.objects.prefetch_related("user").filter(payment_accepted=True)
+    data_list = sorted(([str(p.user), str(p.room) if p.room else ''] for p in prefs),
+                       key=lambda e: last_first_name_key(e[0]))
 
-# https://docs.djangoproject.com/en/1.11/howto/outputting-csv/
-# NOTE: Might not be the best approach - consider using csv module instead
-def csv_response(data, template, filename='file'):
-    response = HttpResponse(content_type='text/csv')
-    response['Content-Disposition'] = 'attachment; filename="{}.csv"'.format(filename)
-    t = loader.get_template(template)
-    c = {'data': data}
-    response.write(t.render(c))
-    return response
+    return csv_response(("User", "Room"), data_list, filename='rooms_by_users')
 
 
 @staff_member_required
 @require_http_methods(['GET'])
-def report(request):
-    zosia = get_object_or_404(Zosia, active=True)
-    rooms = Room.objects.all_visible().prefetch_related('members').all()
-    rooms = sorted(rooms, key=lambda x: str(x))
-    users = UserPreferences.objects.for_zosia(zosia).prefetch_related('user').all()
-    users = sorted(users, key=lambda x: str(x))
-    ctx = {
-        'zosia': zosia,
-        'rooms': rooms,
-        'user_preferences': users
-    }
+def list_by_room(request):
+    def to_key(room_name):
+        room_name = room_name.lower()
+        m = re.match(r"(\D?)(\d*)(.*)", room_name[::-1])
+        g = [x[::-1] for x in reversed(m.groups())]
+
+        return (g[0], int(g[1]), g[2]) if g[1] != '' else (room_name, 0, '')
```

Comment:
You know that I love regexps but are you 100% sure that this solution won't fall if someone will define some strange room names?

Reply:
We'd simply like to sort numbers as integers, not strings. I've already found [this library](https://natsort.readthedocs.io/en/master/index.html), so we can either use it or write the needed part on our own.